src/network/networkd-wiphy.c

```diff
@@ -80,6 +84,188 @@ int wiphy_get_by_name(Manager *manager, const char *name, Wiphy **ret) {
         return 0;
 }
 
+static int link_get_wiphy(Link *link, Wiphy **ret) {
+        _cleanup_(sd_device_unrefp) sd_device *phy = NULL;
+        const char *s;
+        Wiphy *w;
+        int r;
+
+        assert(link);
+        assert(link->manager);
+
+        if (link->iftype != ARPHRD_ETHER)
+                return -EOPNOTSUPP;
+
+        if (!link->sd_device)
+                return -EOPNOTSUPP;
+
+        r = sd_device_get_devtype(link->sd_device, &s);
+        if (r < 0)
+                return r;
+
+        if (!streq_ptr(s, "wlan"))
+                return -EOPNOTSUPP;
+
+        r = sd_device_get_syspath(link->sd_device, &s);
+        if (r < 0)
+                return r;
+
+        s = strjoina(s, "/phy80211");
+        r = sd_device_new_from_syspath(&phy, s);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_sysname(phy, &s);
+        if (r < 0)
+                return r;
+
+        r = wiphy_get_by_name(link->manager, s, &w);
+        if (r < 0)
+                return r;
+
+        /* Optionally assign the sd-device object to the Wiphy object. */
+        if (!w->dev)
+                w->dev = TAKE_PTR(phy);
```

Comment:
This seems a bit weird, wouldn't it be easier to follow if we just attached the phy device object to the Wiphy object immediately after we initialized the wiphy object in `manager_genl_process_nl80211_wiphy()` and only remove it when we free the `Wiphy` object? That makes the lifetime a bit more clear.